geowebcache/core/src/main/java/org/geowebcache/storage/BlobStoreAggregator.java

```diff
@@ -0,0 +1,260 @@
+
+
+
+
+    public BlobStoreAggregator(GridSetBroker gridSetBroker, List<BlobStoreConfiguration> configs) {
+        this.configs = configs == null ? new ArrayList<BlobStoreConfiguration>() : configs;
+        this.gridSetBroker = gridSetBroker;
+        initialize();
+    }
+
+    public BlobStoreAggregator(GridSetBroker gridSetBroker) {
+        this.gridSetBroker = gridSetBroker;
+        reInit();
+    }
+
+    public void addConfiguration(BlobStoreConfiguration config) {
+        initialize(config);
+        List<BlobStoreConfiguration> newList = new ArrayList<BlobStoreConfiguration>(configs);
+        newList.add(config);
+        this.configs = newList;
+    }
+
+    public boolean blobStoreExists(final String blobStoreInfoName) {
+        for (int i = 0; i < configs.size(); i++) {
+            BlobStoreConfiguration configuration = configs.get(i);
+            Optional<BlobStoreInfo> blob = configuration.getBlobStore(blobStoreInfoName);
+            if (blob.isPresent()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns the blob store info named after the {@code blobStoreName} parameter.
+     *
+     * @throws GeoWebCacheException
+     *             if no such blob store info exists
+     */
+    public BlobStoreInfo getBlobStoreInfo(final String blobStoreName) throws GeoWebCacheException {
+        Preconditions.checkNotNull(blobStoreName, "blobStoreName is null");
+
+        for (int i = 0; i < configs.size(); i++) {
+            BlobStoreConfiguration configuration = configs.get(i);
+            Optional<BlobStoreInfo> blob = configuration.getBlobStore(blobStoreName);
+            if (blob.isPresent()) {
+                return blob.get();
+            }
+        }
+        throw new GeoWebCacheException("Thread " + Thread.currentThread().getId()
+                + " Unknown blob store " + blobStoreName + ". Check the logfiles,"
+                + " it may not have loaded properly.");
+    }
+
+    /***
+     * Reinitialization is tricky, because we can't really just lock all the blobstores, because this
+     * would cause people to queue on something that we may not want to exist post reinit.
+     *
+     * So we'll just set the current blobstore set free, ready for garbage collection, and generate a
+     * new one.
+     *
+     */
+    public void reInit() {
+        List<BlobStoreConfiguration> extensions = GeoWebCacheExtensions.extensions(BlobStoreConfiguration.class);
+        this.configs = new ArrayList<BlobStoreConfiguration>(extensions);
+        initialize();
+    }
+
+    public int getBlobStoreCount() {
+        int count = 0;
+        for (int i = 0; i < configs.size(); i++) {
+            BlobStoreConfiguration configuration = configs.get(i);
+            count += configuration.getBlobStoreCount();
+        }
+        return count;
+    }
+
+    public Set<String> getBlobStoreNames() {
+        Set<String> names = new HashSet<String>();
```

Comment:
I'm not sure a set is viable here, as there is nothing enforcing name uniqueness across multiple blobStoreConfigurations